tests/run/i14020.scala

```diff
@@ -0,0 +1,39 @@
+class A:
+  def greeting(name: String = "you") = s"Hello $name"
+
+class A2:
+  inline def greeting(name: String = "you") = s"Hello $name"
+
+class B:
+  val a = A()
+  export a.*
+
+class C:
+  val a = A2()
+  export a.greeting
+
+@main def Test =
+  val b = B()
+
+  println(b.a.greeting())      // works
+  println(b.greeting("John"))  // works
+  println(b.greeting())        // nope !
+
+  val c = C()
+
+  println(c.a.greeting())      // works
+  println(c.greeting("John"))  // works
+  println(c.greeting())        // nope !
+
+  val w = Wolf()
+  import w.given
+
+  println(summon[String]) // error: I found: w.bark(/* missing */summon[String])
+
+
+class Dog:
+  given bark(using msg: String = "Woof!"): String = s"bark: $msg"
```

Comment:
I have found one last issue which is by making `bark` inline:
```scala
class Dog:
   inline given bark(using msg: String = "Woof!"): String = s"bark: $msg"
```
gives
```scala
-- Error: ----------------------------------------------------------------------
31 |summon[String]
   |              ^
   |value dog cannot be accessed as a member of (Wolf_this : (w : Wolf)) from module class i14020$package$.
   | This location contains code that was inlined from i14020.scala:39
1 error found
```

Reply:
you don't get the same error if you try to manually forward to dog.bark in Wolf:
```scala
class Wolf:
  private val dog = Dog()
  inline given bark(using msg: String = "Woof!"): String = dog.bark(using msg)
```